documentation/design-docs/dotnet-tools.md

```diff
@@ -1,124 +1,801 @@
-#Dotnet Diagnostic Tools CLI Design
+# Dotnet Diagnostic Tools CLI Design
 
-Capturing some comments from https://github.com/dotnet/diagnostics/issues/85 as a starting point
-we can edit from. This needs cleanup.
+# User workflows
 
-@davidfowl wrote:
+These are some quick examples of the work we'd expect a .Net developer to want to get a feel for the design. A more complete reference follows.
 
-@shirhatti We're thinking it should be part of dotnet-collect since all of the flags and infrastructure would likely be the same and it should have a "top" like interface. 
+### Ad-hoc Health Monitoring / 1st Level Performance Triage
 
-![image](https://user-images.githubusercontent.com/95136/49297973-09575400-f471-11e8-99ec-823e616eafa2.png)
 
-We'll need to decide what things we show (aggregations and counters)
+**Seeing performance counter values that refresh periodically in-place**
 
+    > dotnet tool install -g dotnet-counters
+    You can invoke the tool using the following command: dotnet-counters
+    Tool 'dotnet-counters' (version '1.0.0') was successfully installed.
+    > dotnet counters monitor --process-id 1902 Microsoft-Windows-DotNETRuntime Microsoft-AspNet
+    
+    Microsoft-Windows-DotNETRuntime: 
+        Total Processor Time (ms)              173923.48 
+        Private Virtual Memory (MB)                 1094       
+        Working Set (MB)                            1982
+        Virtual Memory (MB)                         3041
+        GC Heap Memory (MB)                          784
+        Exception Thrown Rate (exceptions/min)       117
+        Lock Contention Rate (contentions/min)      1792               
+
+     Microsoft-AspNet:
+        Request Rate (requests/sec)                 1915
+        Request Latency (ms)                          34  
+       
+    'p' - pause updates
+    'r' - resume updates
+    'q' - quit
+
+(Counter groups are for example purpose only, exact groups/counters TBD)
+
+### Capture a trace for performance analysis
+
+For analyzing CPU usage, IO, lock contention, allocation rate, etc the investigator wants to capture a performance trace. This trace can then be moved to a developer machine where it can be analyzed with profiling tools such as PerfView/VisualStudio or visualized as a flame graph with speedscope. 
```

Comment:
A little more info (or a link) on speedscope would be useful.

Reply:
👍